[grpc] pkg-config files have `upb` dependencies in the wrong order

**Describe the bug**

Compiling code that uses gRPC via `pkg-config` produces link errors.  The list of `-lupb_*` options seem to be in the wrong order.

**Environment**

- OS: Linux
- Compiler: gcc 12

**To Reproduce**

I am sure this can happen with a smaller repro, but this is how I got it.

1. $HOME/vcpkg-develop/vcpkg install google-cloud-cpp[core,speech]
2. git clone https://github.com/googleapis/google-cloud-cpp
3. cd google-cloud-cpp/google/cloud/speech/quickstart
4. PKG_CONFIG_PATH=$HOME/vcpkg-develop/installed/x64-linux/lib/pkgconfig make
5. See error


```
/usr/bin/ld: /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/lib/pkgconfig/../../lib/libupb.a(decode.c.o): in function `decode_tomsg':
decode.c:(.text+0x1217): undefined reference to `utf8_naive'
/usr/bin/ld: /usr/local/google/home/coryan/vcpkg-develop/installed/x64-linux/lib/pkgconfig/../../lib/libupb.a(decode.c.o): in function `decode_toarray':
decode.c:(.text+0x1a87): undefined reference to `utf8_naive'
collect2: error: ld returned 1 exit status
make: *** [Makefile:36: quickstart] Error 1
```

**Expected behavior**

A successful build.

**Additional context**

When building with CMake the `upb` dependencies are linked in this order:

```
libupb_json.a
libupb_textformat.a 
libupb_reflection.a 
libupb.a 
libupb_fastdecode.a 
libupb_utf8_range.a 
libdescriptor_upb_proto.a
```

Whereas the port for gRPC applies a patch

https://github.com/microsoft/vcpkg/blob/8e3595fad04f60f55250ff0e2f2d4e7e7a580a50/ports/grpc/portfile.cmake#L20

which adds these dependencies to the `pkg-config` file in a different order:

https://github.com/microsoft/vcpkg/blob/8e3595fad04f60f55250ff0e2f2d4e7e7a580a50/ports/grpc/00014-pkgconfig-upbdefs.patch#L10

I think it is a matter of reordering the `upb` libraries in the pkgconfig patch.  I will try to submit a PR to fix this.

--- a/ports/grpc/00014-pkgconfig-upbdefs.patch
+++ b/ports/grpc/00014-pkgconfig-upbdefs.patch
@@ -7,7 +7,7 @@ index 25990a5d8a..4c7831cba5 100644
    "${gRPC_CORE_VERSION}"
    "gpr openssl absl_base absl_bind_front absl_cord absl_core_headers absl_flat_hash_map absl_flat_hash_set absl_function_ref absl_hash absl_inlined_vector absl_memory absl_optional absl_random_random absl_span absl_status absl_statusor absl_str_format absl_strings absl_synchronization absl_time absl_type_traits absl_utility absl_variant"
 -  "-lgrpc -laddress_sorting -lre2 -lupb -lcares -lz"
-+  "-lgrpc -laddress_sorting -lre2 -ldescriptor_upb_proto -lupb_fastdecode -lupb_json -lupb_reflection -lupb_textformat -lupb_utf8_range -lupb -lcares -lz"
++  "-lgrpc -laddress_sorting -lre2 -lupb_json -lupb_textformat -lupb_reflection -lupb -lupb_fastdecode -lupb_utf8_range -ldescriptor_upb_proto -lcares -lz"
    ""
    "grpc.pc")
  

--- a/ports/grpc/vcpkg.json
+++ b/ports/grpc/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "grpc",
   "version-semver": "1.48.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "An RPC library and framework",
   "homepage": "https://github.com/grpc/grpc",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2730,7 +2730,7 @@
     },
     "grpc": {
       "baseline": "1.48.0",
-      "port-version": 1
+      "port-version": 2
     },
     "grppi": {
       "baseline": "0.4.0",

--- a/versions/g-/grpc.json
+++ b/versions/g-/grpc.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "efed7f493dec5fdebaebdec83c521fc860b432a3",
+      "version-semver": "1.48.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "7b2206a9758481b4773a93435aa6cfe7de4bf9e5",
       "version-semver": "1.48.0",
       "port-version": 1

